src/main/java/net/flintmc/gradle/manifest/tasks/GenerateFlintManifestTask.java

```diff
@@ -421,6 +516,18 @@ private InstallInstructionModel buildOwnInstallInstruction() {
       targetPath = "${FLINT_PACKAGE_DIR}/" + getProjectName() + ".jar";
     }
 
+    String sha1Hex;
+    if (manifestType == ManifestType.DISTRIBUTOR) {
+      Jar jar = (Jar) getProject().getTasks().getByName("jar");
+      File singleFile = jar.getOutputs().getFiles().getSingleFile();
+      try (FileInputStream fileInputStream = new FileInputStream(singleFile)) {
+        sha1Hex = Util.sha1Hex(IOUtils.toByteArray(fileInputStream));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      sha1Hex = null;
+    }
```

Comment:
A few issues:
1. This is a bit unreadable, can you maybe add an empty line after the variable declaration?
2. Don't throw runtime exceptions here
3. Add a comment explaining why this logic is required, it isn't obvious from the context